src/model/pitch.rs

```diff
@@ -24,6 +24,25 @@ impl Pitch {
     pub fn all() -> impl Iterator<Item = Pitch> {
         Letter::iter().flat_map(|l| Modifier::iter().map(move |m| Pitch::new(l, m)))
     }
+
+    pub fn index(&self) -> usize {
+        match (self.letter, self.modifier) {
+            (Letter::C, Modifier::Natural) => 0,
+            (Letter::D, Modifier::Flat) => 1,
+            (Letter::D, Modifier::Natural) => 2,
+            (Letter::E, Modifier::Flat) => 3,
+            (Letter::E, Modifier::Natural) => 4,
+            (Letter::F, Modifier::Flat) => 4,
+            (Letter::F, Modifier::Natural) => 5,
+            (Letter::G, Modifier::Flat) => 6,
+            (Letter::G, Modifier::Natural) => 7,
+            (Letter::A, Modifier::Flat) => 8,
+            (Letter::A, Modifier::Natural) => 9,
+            (Letter::B, Modifier::Flat) => 10,
+            (Letter::B, Modifier::Natural) => 11,
+            (Letter::C, Modifier::Flat) => 11,
+        }
+    }
```

Comment:
Some very similar logic exists already in the line parser (`to_wmidi_note`). Could that be updated to use this?

Reply:
Ok yup updated `to_wmidi_note()` to delegate to `pitch.index()` (whose return value clearly wanted to be an `i8` rather than a `usize`)